core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/LruBlockCache.java

```diff
@@ -84,7 +85,7 @@
   static final int statThreadPeriod = 60;
 
   /** Concurrent map (the cache) */
-  private final ConcurrentHashMap<String,CachedBlock> map;
```

Comment:
One more thought.. may want to have an enum cache type.

```java
enum CacheType {
  INDEX,
  DATA,
  SUMMARY
}
```

Then the factory could key use this.

```java
public interface BlockCacheFactory {
  /**
   * TODO document how this can utilize the new arbitrary instance config
   */
  public BlockCache getBlockCache(CacheType ct, AccumuloConfiguration conf);
} 
```

Then the implementation could use this for differentiated config.

```java
public class SuperDuperBlockCacheFactory implement BlockCacheFactory {
   public BlockCache getBlockCache(CacheType ct, AccumuloConfiguration conf) {
       long knob42 = conf.getMemory("general.custom."+ct.name()+".knob42");
        //read whatever is needed from config in local vars
        return new SuperDuperBlockCacheFactory(knob42);
   }
}
```